### ISSUE TITLE
[FEATURE] Publish as Nix package

### ISSUE THREAD

**Issue by geigerj0** [SAP/terraform-provider-btp#984]
### What area do you want to see improved?

terraform provider

### Is your feature request related to a problem? Please describe.

I would like to consume the `terraform-provider-btp` as Nix package to have it available in an OCI image that is being built based on the Nix declarations. 

Currently, it's not yet available https://search.nixos.org/packages?channel=24.11&show=terraform-providers.azurerm&from=0&size=300&sort=alpha_asc&type=packages&query=terraform-providers. 

### Describe the solution you would like

I would love to see `terraform-provider-btp` as Nix package so that users can consume it from within their Nix environments.

**Comment by lechnerc77** [comment]
@geigerj0 Can you kindly point us to where find more information about the steps to provide such a Nix package?

**Comment by geigerj0** [comment]
> [@geigerj0](https://github.com/geigerj0) Can you kindly point us to where find more information about the steps to provide such a Nix package?

I am not quiet familiar so I asked some colleagues, one of them helped out and created this https://github.com/NixOS/nixpkgs/pull/366796 ❤ . The colleague mentioned that someone just needs to take of bumping the version from now and then.

**Comment by yovko** [comment]
Yep. Use `nix flake prefetch github:SAP/terraform-provider-btp/vX.Y.Z` to check the 'hash' value. And [some of those scripts](https://github.tools.sap/cloudfoundry/nixpkgs/tree/main/scripts) to calculate the 'vendorHash' when you need to update the release/version in the future.

**Comment by geigerj0** [comment]
One can consume version `1.8.0` now as Nix package: https://search.nixos.org/packages?channel=unstable&show=terraform-providers.btp. 

I think the only thing that's left is bumping it from time to time

**Comment by lechnerc77** [comment]
Thanks for creating the PR in the nixpkgs repository. We looked into the overall setup and we did not see an easy way to automate the update of the packages once we release a new provider version. Maybe we missed something here.
As we will not do the updates manually in every release, we decided to not support this package ecosystem from our end. 

If we missed something with respect to automating the setup via a GitHub Action in this repository, feel free to contribute a pull request with the corresponding fix.
